processor/schemaprocessor/internal/translation/manager.go

```diff
@@ -0,0 +1,189 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translation // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/translation"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"go.uber.org/zap"
+)
+
+var (
+	errNilValueProvided = errors.New("nil value provided")
+)
+
+type (
+	// Manager is responsible for ensuring that schemas are kept up to date
+	// with the most recent version that are requested.
+	Manager interface {
+		// RequestTranslation will provide either the defined Translation
+		// if it is a known target, or, return a noop variation.
+		// In the event that a matched Translation, on a missed version
+		// there is a potential to block during this process.
+		// Otherwise, the translation will allow concurrent reads.
+		RequestTranslation(ctx context.Context, schemaURL string) Translation
```

Comment:
My only thought so far is what would happen if there is a new published version for the Translation to fetch?

I suspect this will make it significantly easier to manage translations but I was worried about potential race conditions in this approach if types were incorrectly set for this.